File: ios/SsdidWallet/Domain/Model/Identity.swift
import Foundation

struct Identity: Codable, Identifiable {
    let name: String
    let did: String
    let keyId: String
    let algorithm: Algorithm
    let publicKeyMultibase: String
    let createdAt: String
    var isActive: Bool = true
    var recoveryKeyId: String? = nil
    var hasRecoveryKey: Bool = false
    var preRotatedKeyId: String? = nil

    var id: String { keyId }
}

File: android/sdk/ssdid-core/src/main/java/my/ssdid/sdk/domain/model/Identity.kt
package my.ssdid.sdk.domain.model

import kotlinx.serialization.Serializable

@Serializable
data class Identity(
    val name: String,
    val did: String,
    val keyId: String,
    val algorithm: Algorithm,
    val publicKeyMultibase: String,
    val createdAt: String,
    val isActive: Boolean = true,
    val recoveryKeyId: String? = null,
    val hasRecoveryKey: Boolean = false,
    val preRotatedKeyId: String? = null,
    val profileName: String? = null,
    val email: String? = null,
    val emailVerified: Boolean = false
) {
    fun claimsMap(): Map<String, String> = buildMap {
        profileName?.let { put("name", it) }
        email?.let { put("email", it) }
    }
}
